### PR TITLE
Add changelog for `4.35.0`

### PR DESCRIPTION
**Pull request by rileykarson**

### CODE REVIEW

**Review by melinath — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,38 @@
...
+* compute: added support documentation links to error messages for certain Compute Operation errors. ([#12418](https://github.com/hashicorp/terraform-provider-google/pull/12418))
+* container: added `service_external_ips_config` support to `cluster_container` resource. ([#12415](https://github.com/hashicorp/terraform-provider-google/pull/12415))
+* container: added `enable_cost_allocation` to `google_container_cluster` ([#12416](https://github.com/hashicorp/terraform-provider-google/pull/12416))
+* dns: added `behavior` field to `google_dns_response_policy_rule` resource ([#12407](https://github.com/hashicorp/terraform-provider-google/pull/12407))
+* monitoring: added `force_delete` field to `google_monitoring_notification_channel` resource ([#12414](https://github.com/hashicorp/terraform-provider-google/pull/12414))
+* vertexai: added `encryption_spec` field to `google_vertex_ai_featurestore` resource (beta) ([#12430](https://github.com/hashicorp/terraform-provider-google/pull/12430))
```
Comment (megan07) [inline CHANGELOG.md:13]: this looks to be beta

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,38 @@
...
+NOTES:
+* updated Bigtable go client version from 1.13 to 1.16. ([#12349](https://github.com/hashicorp/terraform-provider-google/pull/12349))
+
+IMPROVEMENTS:
+* apigee: added support for specifying retention when deleting `google_apigee_organization` ([#12336](https://github.com/hashicorp/terraform-provider-google/pull/12336))
+* appengine: added `app_engine_apis` field to `google_app_engine_standard_app_version` resource ([#12339](https://github.com/hashicorp/terraform-provider-google/pull/12339))
+* cloudfunction2: promoted to `google_cloudfunctions2_function` ga ([#12322](https://github.com/hashicorp/terraform-provider-google/pull/12322))
+* compute: improved error messaging for compute errors ([#12333](https://github.com/hashicorp/terraform-provider-google/pull/12333))
+* container: added general field `reservation_affinity` to `google_container_node_pool` ([#12375](https://github.com/hashicorp/terraform-provider-google/pull/12375))
+* container: added field `auto_provisioning_network_tags` to `google_container_cluster` (beta) ([#12347](https://github.com/hashicorp/terraform-provider-google/pull/12347))
```
Comment (megan07) [inline CHANGELOG.md:29]: this also looks to be beta

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,38 @@
...
+* sql: added support for major version upgrade to `google_sql_database_instance ` resource ([#12338](https://github.com/hashicorp/terraform-provider-google/pull/12338))
+
+BUG FIXES:
+* bigtable: fixed comparing column family name when reading a GC policy. ([#12381](https://github.com/hashicorp/terraform-provider-google/pull/12381))
+* bigtable: passed `isTopeLevel` in getGCPolicyFromJSON() instead of hardcoding it to true. ([#12351](https://github.com/hashicorp/terraform-provider-google/pull/12351))
```
Comment (megan07) [inline CHANGELOG.md:34]: is this message relevant for the changelog - do we want something more useful for the user?

File: CHANGELOG.md (hunk 2 above)
Comment (rileykarson) [reply CHANGELOG.md:29]: This is from last release- leaving unchanged, as it matches HEAD

File: CHANGELOG.md (hunk 3 above)
Comment (rileykarson) [reply CHANGELOG.md:34]: This is from last release- leaving unchanged, as it matches HEAD

File: CHANGELOG.md (hunk 1 above)
Comment (rileykarson) [reply CHANGELOG.md]: Done

**Pull request merged**